lib/msal-common/src/cache/entities/CredentialEntity.ts

```diff
@@ -108,17 +108,17 @@ export class CredentialEntity {
      * @param key
      */
     static getCredentialType(key: string): string {
-        // First keyword search will match all "AccessToken" and "AccessToken_With_AuthScheme" credentials
-        if (key.indexOf(CredentialType.ACCESS_TOKEN.toLowerCase()) !== -1) {
-            // Perform second search to differentiate between "AccessToken" and "AccessToken_With_AuthScheme" credential types
-            if (key.indexOf(CredentialType.ACCESS_TOKEN_WITH_AUTH_SCHEME.toLowerCase()) !== -1) {
-                return CredentialType.ACCESS_TOKEN_WITH_AUTH_SCHEME;
+        const separator = Separators.CACHE_KEY_SEPARATOR;
+        const domainRe = "(https?:\\/\\/)?([\\w-]+\\.)*([\\w-]{1,63})(\\.(\\w{2,3}))(\\:[0-9]{4,5})?";
+
+        for (const credType of Object.entries(CredentialType)) {
+            const credKey = credType[0];
+            const credVal = credType[1].toLowerCase();
+            // Verify credential type is preceded by a valid host name (environment)
+            // eslint-disable-next-line security/detect-non-literal-regexp
```

Comment:
This is a rule we might not get away with disabling as the code which violates this rule will likely get flagged in the CodeQL scans. Is there a different way we can do this?

Reply:
Fixed.